providers/src/airflow/providers/apache/kafka/hooks/consume.py

```diff
@@ -36,6 +48,7 @@ def __init__(self, topics: Sequence[str], kafka_config_id=KafkaBaseHook.default_
         self.topics = topics
 
     def _get_client(self, config) -> Consumer:
+        config["error_cb"] = error_callback
```

Comment:
It might not be the best practice to update user-provided data. As an alternative, we could create a shallow copy instead.

Additionally, we can assume that the user has already provided an error_cb. In this case, we need to decide how to handle this behavior, which is an open question. One possible solution is to call the user-defined error_cb first and then invoke our custom implementation if the user does not raise an exception. This approach makes sense because authentication errors are not recoverable, so they should interrupt execution.

Reply:
you approach is more robust, but if user defined error_cb does not raise error, the operator will succeed instead of failing. i am thinking of wrapping the function to make it predictable

Reply:
> Additionally, we can assume that the user has already provided an error_cb.

`config` may already contain handler. In this case you will override user defined handler.

Reply:
@SuccessMoses , they can just specify the import path. Similar to how we take `apply_function`

Reply:
ok